File: backend/src/PayTerminal.Application/DTOs/Requests/RegisterTerminalRequest.cs
using System.ComponentModel.DataAnnotations;

namespace PayTerminal.Application.DTOs.Requests;

public class RegisterTerminalRequest
{
    [Required]
    public Guid MerchantId { get; set; }

    [Required, StringLength(10)]
    public string PairingCode { get; set; } = string.Empty;

    [Required, MaxLength(120)]
    public string Name { get; set; } = string.Empty;
}

File: app/src/main/java/com/arafath/payterminalversion2/data/remote/dto/RegisterTerminalRequest.java
package com.arafath.payterminalversion2.data.remote.dto;

public class RegisterTerminalRequest {
    public String merchantId;
    public String pairingCode;
    public String name;

    public RegisterTerminalRequest(String merchantId, String pairingCode, String name) {
        this.merchantId = merchantId;
        this.pairingCode = pairingCode;
        this.name = name;
    }
}